types/convert.go

```diff
@@ -443,12 +443,15 @@ func floatStrToIntStr(sc *stmtctx.StatementContext, validFloat string, oriStr st
 	if err != nil {
 		return validFloat, errors.Trace(err)
 	}
-	if exp > 0 && int64(intCnt) > (math.MaxInt64-int64(exp)) {
-		// (exp + incCnt) overflows MaxInt64.
+	intCnt += exp
+	// intCnt will < 0 when overflow
+	if intCnt < 0 || intCnt > 20 {
```

Comment:
The `StrToUint` will use this function to extract valid int string prefix, so I think we cannot limit the length to `20`.